lib/input_file_converter.py

```diff
@@ -265,7 +265,7 @@ def execute(self):
                     gene_name = transcript['SYMBOL']
                     index = '%s.%s.%s.%s' % (gene_name, transcript_name, consequence, amino_acid_change_position)
                     if index in indexes:
-                        sys.exit("TSV index already exists")
+                        print("Warning: TSV index already exists: {}".format(index))
```

Comment:
We should not change this to just a warning. Downstream steps in the main pVACseq pipeline require that the TSV index be unique between the different variants so that we can use it as a unique identifier. Would you be able to share some example data where you're running into this error?

Reply:
Susanna,
Unfortunately, I cannot share my input VCFs. I was running this on internal proprietary data. Mutect2 WES VCF. Only one or two entries per sample were generating this error, so I did not have any choice but to turn this into a warning.
I thought that the step in `generate_protein_fasta.py` was the last in the pipeline, with no downstream analysis afterwards? I need these peptides for building a DB for proteomics peptide identification.
The code in `generate_protein_fasta.py` appears to be largely copy/pasted from `generate_fasta.py`. The latter one is probably that is used by downstream steps? Maybe then have an option in `input_file_converter.py` that turns exit into a warning when called only from `generate_protein_fasta.py`?

Reply:
`tools/pvacseq/generate_protein_fasta.py` is a standalone script that just generates the peptide fasta and is invoked by the `pvacseq generate_protein_fasta` command. The full `pvacseq run` command generates the protein fasta, makes binding predictions, and filters the output. `lib/generate_fasta.py` is a remnant back from the pVAC-Seq repository before we moved the codebase to pVACtools and reorganized everything. The code now lives in the FastaGenerator class (`lib/fasta_generator.py`). `lib/generate_fasta.py` should actually be deleted. You will see that `generate_protein_fasta.py` is command is using this class.

With your change what would happen is that you are potentially losing peptides because the same index/identifier points to two different peptide sequences and the first one gets overwritten in the `parse_files` method of `generate_protein_fasta.py`.

Would it be possible to come up with dummy data that recreates this error? The only situation I can think of where this might be happening is if there are two variants in your VCF with the same position that also have the same consequence. In that case I think your solution would work because the resulting peptide sequences between the two variants would be the same but I would want to make sure that we aren't creating a scenario where we are losing peptides.

Reply:
Susanna,
Attached if a single variant from a model cell culture that triggers the "duplicated index" condition. This single-variant file is enough:
```
$ pvacseq generate_protein_fasta scratch/dup_index.vcf 60 scratch/tmp.fasta
Converting VCF to TSV
Warning: TSV index already exists: CD34.ENST00000367036.missense.28S/F. Entry is: Record(CHROM=1, POS=208063100, REF=G, ALT=[A, GA])
Completed
Generating Variant Peptide FASTA and Key File
Completed
Parsing the Variant Peptide FASTA and Key File
Completed
```
Also attached is the same variant before VEP annotation, to see what has been added by VEP. It looks like the VEP annotation contain multiple transcripts triggering the condition.
I will leave it up to you to modify my changes as you see fit.
Thanks,
Andrey

[duplicated_index_test_data.zip](https://github.com/griffithlab/pVACtools/files/1782494/duplicated_index_test_data.zip)

Reply:
The bug that you were seeing with your example is that pVACseq is interpreting the VEP allele incorrectly for this case. Your example has two alternate alleles, `A` and `GA` with a `G` reference with corresponding alleles in the VEP CSQ field. However, if your entry only had the `GA` alternate allele, VEP would "normalize" this as an `A` allele, which is what pVACseq is looking at. Since in your example `A` actually exists in the CSQ field, pVACseq looks at the `A` VEP entries for both alternate alleles. I will figure out a way to handle this case.